File: tracks/typescript/exercises/08-to-the-cloud/worker.ts
/**
 * The same Worker as exercise 2, connected to Temporal Cloud.
 * Workflow code and Task Queue configuration are unchanged.
 */

import { NativeConnection, Worker } from '@temporalio/worker';

const REQUIRED = [
  'TEMPORAL_ADDRESS',
  'TEMPORAL_NAMESPACE',
  'TEMPORAL_API_KEY',
] as const;

async function main(): Promise<void> {
  const missing = REQUIRED.filter((name) => !process.env[name]);
  if (missing.length > 0) {
    throw new Error(`Set ${missing.join(', ')} first — see this exercise's README.`);
  }

  const address = process.env.TEMPORAL_ADDRESS as string;
  const namespace = process.env.TEMPORAL_NAMESPACE as string;
  const apiKey = process.env.TEMPORAL_API_KEY as string;
  const connection = await NativeConnection.connect({
    address,
    tls: true,
    apiKey,
  });

  try {
    const worker = await Worker.create({
      connection,
      namespace,
      workflowsPath: require.resolve('./workflows'),
      taskQueue: 'tagline-tasks',
    });
    console.log(
      `Worker connected to ${namespace} on Temporal Cloud. Ctrl-C to stop.`,
    );
    await worker.run();
  } finally {
    await connection.close();
  }
}

main().catch((error: unknown) => {
  console.error(error);
  process.exit(1);
});

File: tracks/python/exercises/08-to-the-cloud/worker.py
"""The same worker you ran all workshop — pointed at Temporal Cloud.

Notice what changed compared to exercises/02-first-workflow/solution/worker.py:
the Client.connect() call. Nothing else. Workflow code, task queues, CLI
commands — all identical. That's the dev-server-to-production story.
"""

import asyncio
import os
import sys

from temporalio.client import Client
from temporalio.worker import Worker

from workflows import TaglineWorkflow

REQUIRED = ("TEMPORAL_ADDRESS", "TEMPORAL_NAMESPACE", "TEMPORAL_API_KEY")


async def main():
    missing = [k for k in REQUIRED if not os.environ.get(k)]
    if missing:
        sys.exit(f"Set {', '.join(missing)} first — see the README in this directory.")

    client = await Client.connect(
        os.environ["TEMPORAL_ADDRESS"],        # regional endpoint, e.g. us-east-1.aws.api.temporal.io:7233
        namespace=os.environ["TEMPORAL_NAMESPACE"],  # <namespace>.<account-id>
        api_key=os.environ["TEMPORAL_API_KEY"],
        tls=True,                              # Cloud always speaks TLS
    )
    worker = Worker(
        client,
        task_queue="tagline-tasks",
        workflows=[TaglineWorkflow],
    )
    print(f"Worker connected to {os.environ['TEMPORAL_NAMESPACE']} on Temporal Cloud. Ctrl-C to stop.")
    await worker.run()


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        print("Worker stopped.")
